design-documents/asynchronous-operations/poison-pill.md

```diff
@@ -0,0 +1,20 @@
+## Glossary
+
+poison pill
+: message in Data Storage that signals that consumer should die
+
+### Overview
+
+Right now our consumer depends on config/objects in-memory state. After application was bootstrapped it initialize a lot of object that are always kept in memory after first load - config, websites, etc.., since our consumers are long-living and their live time limited only by message count it leads to wrong im-memory config/objects states and we need to introduce the way to reinit them.
+
+### Design
+
+Before process new message consumer should ask Data Storage if there are poison pill for him. If yes, consumer dies and next cron:run will start new instance of consumer.
```

Comment:
How a consumer will communicate with data storage?

Reply:
I think yeas, we are going to introduce the basic implementation inside our framework and other modules will use it

Reply:
Could you add more details about API which should be introduced?